### ISSUE TITLE
`KeyError` when using docparams and NotImplementedError is documented

### ISSUE THREAD

**Issue by adamtheturtle** [PyCQA/pylint#2102]
Hello and thank you for the software.
In the most recent update (1.9.0) of `pylint`, a bug was introduced that means that legitimate documentation of `NotImplementedError` can cause a `KeyError` to be raised when using the `docparams` plugin.

### Steps to reproduce
1. Set `load-plugins=pylint.extensions.docparams` in `.pylintrc`:

```
(e2e) Adam@MacBook-Pro ~/D/pylint-experiment> cat .pylintrc 
[MASTER]

load-plugins=pylint.extensions.docparams
```

2. Document that a `NotImplementedError` is raised:

```
(e2e) Adam@MacBook-Pro ~/D/pylint-experiment> cat example.py 
def foo():
    """
    Raises:
        NotImplementedError: When called.
    """
    raise NotImplementedError
```

3. Run `pylint`

### Current behavior

A `KeyError` is raised.

```
(e2e) Adam@MacBook-Pro ~/D/pylint-experiment> pylint example.py 
Using config file /Users/Adam/Desktop/pylint-experiment/.pylintrc
************* Module example
C:  1, 0: Missing module docstring (missing-docstring)
C:  1, 0: Black listed name "foo" (blacklisted-name)
Traceback (most recent call last):
  File "/Users/Adam/.virtualenvs/e2e/bin/pylint", line 11, in <module>
    sys.exit(run_pylint())
  File "/Users/Adam/.virtualenvs/e2e/lib/python3.6/site-packages/pylint/__init__.py", line 16, in run_pylint
    Run(sys.argv[1:])
  File "/Users/Adam/.virtualenvs/e2e/lib/python3.6/site-packages/pylint/lint.py", line 1347, in __init__
    linter.check(args)
  File "/Users/Adam/.virtualenvs/e2e/lib/python3.6/site-packages/pylint/lint.py", line 768, in check
    self._do_check(files_or_modules)
  File "/Users/Adam/.virtualenvs/e2e/lib/python3.6/site-packages/pylint/lint.py", line 901, in _do_check
    self.check_astroid_module(ast_node, walker, rawcheckers, tokencheckers)
  File "/Users/Adam/.virtualenvs/e2e/lib/python3.6/site-packages/pylint/lint.py", line 980, in check_astroid_module
    walker.walk(ast_node)
  File "/Users/Adam/.virtualenvs/e2e/lib/python3.6/site-packages/pylint/utils.py", line 1014, in walk
    self.walk(child)
  File "/Users/Adam/.virtualenvs/e2e/lib/python3.6/site-packages/pylint/utils.py", line 1014, in walk
    self.walk(child)
  File "/Users/Adam/.virtualenvs/e2e/lib/python3.6/site-packages/pylint/utils.py", line 1011, in walk
    cb(astroid)
  File "/Users/Adam/.virtualenvs/e2e/lib/python3.6/site-packages/pylint/extensions/docparams.py", line 217, in visit_raise
    self._add_raise_message(missing_excs, func_node)
  File "/Users/Adam/.virtualenvs/e2e/lib/python3.6/site-packages/pylint/extensions/docparams.py", line 425, in _add_raise_message
    missing_excs.remove('NotImplementedError')
KeyError: 'NotImplementedError'
```

### Expected behavior

No error is shown.

### pylint --version output

```
(e2e) Adam@MacBook-Pro ~/D/pylint-experiment> pylint --version
Using config file /Users/Adam/Desktop/pylint-experiment/.pylintrc
pylint 1.9.0, 
astroid 1.6.4
Python 3.6.5 (default, Mar 30 2018, 06:41:53) 
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```
